workflow/src/legenddataflow/FileKey.py

```diff
@@ -101,9 +108,21 @@ def parse_keypart(cls, keypart):
         return cls(**d)
 
     def expand(self, file_pattern, **kwargs):
+        if isinstance(file_pattern, Path):
+            file_pattern = file_pattern.as_posix()
```

Comment:
I think this should also be robust:

```suggestion
        file_pattern = str(file_pattern)
```